Add SQLite `KVStore` implementation

tnull opened the pull request:
Based on #2472.

We add an `SQLite` based implementation for the newly upstreamed `KVStore` interface.

tnull committed:
Add `SqliteStore` implementation

We upstream our `SqliteStore` implementation that allows persistence
towards an SQLite database backend.

TheBlueMatt reviewed (commented):
MSRV isn't a huge deal, just need to feature-gate it and test a relevant MSRV for the feature (does rustsqlite provide some MSRV guarantees or does it just break compat at will?)

tnull committed:
Add `TestSyncStore`

We add a `KVStore` implementation that wraps all three of our `KVStore`
implementation and uses them in lock step while asserting they behave
identical. This will be useful for (upcoming) integration tests.

tnull committed:
f Add schema migrations and test them

tnull committed:
f Account for renamed namespaces

tnull committed:
f Make more robust and account for renamed namespaces
